server/sonar-server/src/main/java/org/sonar/server/settings/ws/ValuesAction.java

```diff
@@ -181,18 +190,18 @@ ValuesWsResponse build() {
 
     private void processSettings() {
       settings.forEach(setting -> {
-        Settings.Setting.Builder valueBuilder = getOrCreateValueBuilder(setting.getKey());
+        Settings.Setting.Builder valueBuilder = getOrCreateValueBuilder(deprecatedKeysByValidKeys.get(setting.getKey()));
         valueBuilder.setDefault(setting.isDefault());
         setInherited(setting, valueBuilder);
         setValue(setting, valueBuilder);
       });
     }
 
-    private Settings.Setting.Builder getOrCreateValueBuilder(String propertyKey) {
-      Settings.Setting.Builder valueBuilder = settingsBuilderByKey.get(propertyKey);
+    private Settings.Setting.Builder getOrCreateValueBuilder(String key) {
+      Settings.Setting.Builder valueBuilder = settingsBuilderByKey.get(key);
```

Comment:
Because the inheritance is handled here : first default settings are created, then global ones, then module ones. 
If a module setting overrides a global setting, then the valueBuilder already exists for that key.